c/frame/names.cc

```diff
@@ -139,6 +139,10 @@ name: str
 
 oobj Frame::colindex(const PKArgs& args) {
   auto col = args[0];
+  if (!col) {
```

Comment:
Do you think there is a way to have this kind of check being automatic with some default message for all the positional parameters?

Reply:
Positional arguments may still be optional, for example consider `range(a) / range(a, b) / range(a, b, c)`. Since we do not "bake" the defaults into the function's signature, it is the application that must inspect whether the arguments where there or not.

That said, we have `args.get<T>(pos)` and `args.get<T>(pos, default)` helper functions to deal with simple cases.

Reply:
Yes, sure. May be it is not related to positional/keyword parameters at all, but just to all the parameters. When we specify a list of parameters in `PKArgs`, we can mark a parameter as a required or as an optional one. Then for required parameters the check is done automatically.

Reply:
How would the API look like?

Reply:
May be using `std::map` for defining a list of parameters and specifying if they're required or optional:

```
static PKArgs args_range(
    3, 0, 0, false, false,
    {{"a", PKArgs::REQUIRED}, {"b", PKArgs::OPTIONAL}, {"c", PKArgs::OPTIONAL}}, "range",
    ...
```

then, before the `range` method is called, we walk through this map and do checks for all the `PKArgs::REQUIRED` parameters. Is it possible to be done with the way how `ADD_METHOD` and `ExtType<T>::Methods` work?

Reply:
Well, in that case you could simply pass the number of required arguments (owing to the fact that the required argument are always in front).

Reply:
Yep, one can also re-arrange them and pass the number of required parameters. However, if there is some other types, like PKArgs::REQUIRED_NOT_NONE, or something like that, having a map just makes it more flexible.